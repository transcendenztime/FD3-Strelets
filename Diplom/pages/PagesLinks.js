import React from 'react';
import { NavLink } from 'react-router-dom';

import CategoriesLinks from './CategoriesLinks';


import '../styles/pages/PagesLinks.css';

class PagesLinks extends React.Component {
          
  render() {

    return (
      <ul className = "main-menu">
        <li><NavLink to="/" exact className="PageLink" activeClassName="ActivePageLink">Главная</NavLink></li>
        <li>
          <NavLink to="/catalogue" className="PageLink catalogueLink" activeClassName="ActivePageLink" onClick = {this.removeFilter}>
            Каталог
          </NavLink>
          {<CategoriesLinks />}
        </li>
        <li><NavLink to="/warranty" className="PageLink" activeClassName="ActivePageLink">Гарантии</NavLink></li>
        <li><NavLink to="/partnership" className="PageLink" activeClassName="ActivePageLink">Сотрудничество</NavLink></li>
        <li><NavLink to="/contacts" className="PageLink" activeClassName="ActivePageLink">Контакты</NavLink></li>
        <li><NavLink to="/cart" className="PageLink" activeClassName="ActivePageLink">Корзина</NavLink></li>
      </ul>
    );
    
  }

}
     
export default PagesLinks;
    