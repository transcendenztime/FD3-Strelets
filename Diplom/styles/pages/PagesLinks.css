.main-menu {    
    list-style-type: none;
}


.PageLink {    
	color: white;
    text-decoration: none;
    padding: 0 20px;
    height: 50px;
    line-height: 50px;
}

.PageLink:hover {
    animation: PageLink-hover .5s forwards;
}
@keyframes PageLink-hover {
    0% {
        color: white;
    }
    100% {
        color: #112725;
    }
}

.ActivePageLink {
    font-weight: bold;
    color: #112725;
}