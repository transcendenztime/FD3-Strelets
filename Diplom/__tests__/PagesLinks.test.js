import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import PagesLinks from '../pages/PagesLinks';

test('внешний вид PagesLinks', () => {

    // создаём тестовую версию компонента
    const component = renderer.create(
      <Router>
          <PagesLinks />
      </Router>
    );
  
    // получаем снэпшот (HTML-снимок) компонента для сверки, что вёрстка не испортилась
    let componentTree=component.toJSON();
    expect(componentTree).toMatchSnapshot();
          
  });