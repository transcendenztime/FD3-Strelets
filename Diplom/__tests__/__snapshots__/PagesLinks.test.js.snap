// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`внешний вид PagesLinks 1`] = `
<ul
  className="main-menu"
>
  <li>
    <a
      aria-current={null}
      className="PageLink"
      href="/"
      onClick={[Function]}
      style={undefined}
    >
      Главная
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="PageLink catalogueLink"
      href="/catalogue"
      onClick={[Function]}
      style={undefined}
    >
      Каталог
    </a>
    <ul
      className="category-menu main-menu"
    >
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/semena-ovoshchey"
          onClick={[Function]}
          style={undefined}
        >
          Семена овощей
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/semena-cvetov"
          onClick={[Function]}
          style={undefined}
        >
          Семена цветов
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/sazhency-roz"
          onClick={[Function]}
          style={undefined}
        >
          Саженцы роз
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/lukovicy-i-klubnelukovicy-cvetov"
          onClick={[Function]}
          style={undefined}
        >
          Луковицы и клубнелуковицы цветов
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/plodovye-derevya-i-kustarniki"
          onClick={[Function]}
          style={undefined}
        >
          Плодовые деревья и кустарники
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/hvoynye-rasteniya"
          onClick={[Function]}
          style={undefined}
        >
          Хвойные растения
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/dekorativnye-kustarniki"
          onClick={[Function]}
          style={undefined}
        >
          Декоративные кустарники
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="CategoryLink PageLink"
          href="/catalogue/rododendrony"
          onClick={[Function]}
          style={undefined}
        >
          Рододендроны
        </a>
      </li>
    </ul>
  </li>
  <li>
    <a
      aria-current={null}
      className="PageLink"
      href="/warranty"
      onClick={[Function]}
      style={undefined}
    >
      Гарантии
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="PageLink"
      href="/partnership"
      onClick={[Function]}
      style={undefined}
    >
      Сотрудничество
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="PageLink"
      href="/contacts"
      onClick={[Function]}
      style={undefined}
    >
      Контакты
    </a>
  </li>
  <li>
    <a
      aria-current={null}
      className="PageLink"
      href="/cart"
      onClick={[Function]}
      style={undefined}
    >
      Корзина
    </a>
  </li>
</ul>
`;
